src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/DataObject.DataStore.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.IO;
+using System.Runtime.Serialization;
+using System.Windows.Media.Imaging;
+using MS.Internal;
+
+namespace System.Windows;
+
+public sealed partial class DataObject
+{
+    private partial class DataStore : IDataObject
+    {
+        // Data hash table.
+        private readonly Dictionary<string, DataStoreEntry> _data = [];
```

Comment:
Since we are looking into making the index public parameter, why would you remove it from DataStoreEntry and make the GetData ignore it? We will have to add it back to support it.

Reply:
Because it has to be redone in the shared code anyway. Right now, it does nothing and the shared code has no such concept- this facilitates the move to the shared code.